### PR TITLE
make before_suite/after_suite to deprecated

### PR DESCRIPTION
**Pull request by LiliDeng**

### CODE REVIEW

**Comment by squirrelsc** [comment]
Please make sure it's not used in our code.

File: lisa/testsuite.py [hunk 1]
```diff
@@ -539,10 +540,18 @@ def __init__(
...
-        ...
+        warnings.warn(
+            "before_suite is deprecated, use before_case instead",
+            DeprecationWarning,
+            stacklevel=2,
+        )
 
     def after_suite(self, log: Logger, **kwargs: Any) -> None:
-        ...
+        warnings.warn(
```
Comment (squirrelsc) [inline lisa/testsuite.py]: deprecate is an annotation on methods, so it can raise error when writing code, instead of runtime.

**Comment by LiliDeng** [comment]
> Please make sure it's not used in our code.

updated

File: lisa/testsuite.py (hunk 1 above)
Comment (LiliDeng) [reply lisa/testsuite.py]: I didn't figure out a way that can give warning or error when user is writing code without running the code yet.

Comment (squirrelsc) [reply lisa/testsuite.py]: I just found out the "deprecated" is not a standard lib. We can continue to use `warnings.warn`.

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**